moved FAQ data to _data directory (#45)

Tilak999 opened the pull request:
Moved the faq data to faq.yml, this will make adding FAQs easier.

bhansa commented:
Thanks @Tilak999 for your contribution.

ananyo2012 commented on faq.md:
```diff
@@ -3,19 +3,7 @@ layout: flexible
-
-##### Q. I am a speaker/volunteer at the conference. Do I need to buy a ticket?
-Yes. At PyConf, everybody buys tickets. Even the event organizers buy tickets. Your ticket helps keep the event affordable for all.
+{% for faqItem in site.data.faq %}
```
Since we are using yml files for extracting data now lets put this in a html file and use the usual html markup tags. Also introduce the [Q & A schema](https://schema.org/Question).

tilak999 replied:
Q&A schema has lot of attributes and for faq we don't need alot of them so could you provide the attributes that we should include.

tilak999 replied:
Also to add, questions and answers are currectly have markdown syntax. To convert the file into HTML, we would have to replace markdown syntax with HTML tags.

ananyo2012 replied:
> Q&A schema has lot of attributes and for faq we don't need alot of them so could you provide the attributes that we should include.

@Tilak999 Just the Question and Answer attribute will be enough

> Also to add, questions and answers are currectly have markdown syntax. To convert the file into HTML, we would have to replace markdown syntax with HTML tags.

Yes replace the .md file with .html file. Since we will pull data from `_data` folder we can update to html now.

tilak999 replied:
made the changes, please review.

tilak999 commented:
updated the faq.yml as per QA Schema and also added script tag used by google search indexing for showing FAQ in search directly.

[Reference to google FAQ schema](https://developers.google.com/search/docs/data-types/faqpage)

bhansa commented on faq.html:
```diff
@@ -0,0 +1,29 @@
+        "@type": "Answer",
+        "text": "{{faqItem.acceptedAnswer.text}}"
+      }
+    }
```
Add delimiter `,`  for json data.

bhansa replied:
@Tilak999 In case you have missed this comment, please add the delimiter otherwise your json data will become invalid schema.

bhansa replied:
Also make sure the `mainEntity` contains a valid value.

tilak999 replied:
Thanks for pointing out the issue, i have updated `faq.html` to properly render the JSON data and validated the same using the tool link provided by you.

ananyo2012 commented on _data/faq.yml:
```diff
@@ -0,0 +1,37 @@
+"@context": https://schema.org
+"@type": FAQPage
+mainEntity:
+- "@type": Question
```
The additional `@type` field can be removed since they don't seemed to be used anywhere. Also you can simply use `question` and `answer` fields instead of creating the nested field

tilak999 replied:
Those fields are required to follow the QA schema and google search indexing needs the FAQ page in proper QA schema.

I recommend that we follow QA schema for better search visibility on google.

ananyo2012 replied:
I get that, indeed you are doing that in the json-ld schema. But why do we need them in the yml file ? These field values will remain same through out the schema so you don't need to put them in the yml file.

tilak999 replied:
Okay, i get your point. I will update the `faq.yml` file to only contain Q & A

bhansa commented:
@Tilak999  looks good. @ananyo2012  Merging this in.

ananyo2012 commented:
Thanks @Tilak999 . This looks great :tada: Please have a look at other issues as well. And good work @bhansa finding the schema bug in the change. :smiley:

ananyo2012 commented:
@all-contributors Please add @Tilak999 for code
